java/client/src/main/java/glide/api/logging/Logger.java

```diff
@@ -0,0 +1,191 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.logging;
+
+import static glide.ffi.resolvers.LoggerResolver.initInternal;
+import static glide.ffi.resolvers.LoggerResolver.logInternal;
+
+import lombok.Getter;
+import lombok.NonNull;
+
+/**
+ * A singleton class that allows logging which is consistent with logs from the internal rust core.
+ * The logger can be set up in 2 ways -
+ *
+ * <ol>
+ *   <li>By calling <code>Logger.init</code>, which configures the logger only if it wasn't
+ *       previously configured.
+ *   <li>By calling <code>Logger.setLoggerConfig</code>, which replaces the existing configuration,
+ *       and means that new logs will not be saved with the logs that were sent before the call.
+ * </ol>
+ *
+ * If <code>setLoggerConfig</code> wasn't called, the first log attempt will initialize a new logger
+ * with default configuration decided by Glide core.
+ */
+public final class Logger {
+    @Getter
+    public enum Level {
+        DISABLED(-2),
+        DEFAULT(-1),
+        ERROR(0),
+        WARN(1),
+        INFO(2),
+        DEBUG(3),
+        TRACE(4);
+
+        private final int level;
+
+        Level(int level) {
+            this.level = level;
+        }
+
+        public static Level fromInt(int i) {
+            switch (i) {
+                case 0:
+                    return ERROR;
+                case 1:
+                    return WARN;
+                case 2:
+                    return INFO;
+                case 3:
+                    return DEBUG;
+                case 4:
+                    return TRACE;
+                default:
+                    return DEFAULT;
+            }
+        }
+    }
+
+    @Getter private static Level loggerLevel;
+
+    private static void initLogger(@NonNull Level level, String fileName) {
+        if (level == Level.DISABLED) {
```

Comment:
I've started work on this. It's not ready yet and probably won't be in until post-GA, but there's a draft PR up here: https://github.com/Bit-Quill/glide-for-redis/pull/399

Reply:
DISABLED option was added to Java side for testing, since we don't want to log when we have mocks.  There's other ways to do it (mocking static objects is annoying, but possible), but this is the best way.

Reply:
@jonathanl-bq adding it to redis-rs isn't urgent, you can open an issue and we'll do it post GA